src/main/java/at/hannibal2/skyhanni/features/skillprogress/SkillProgress.kt

```diff
@@ -0,0 +1,458 @@
+            }
+
+            SkillProgressConfig.TextAlignment.CENTERED,
+            SkillProgressConfig.TextAlignment.LEFT,
+            SkillProgressConfig.TextAlignment.RIGHT -> {
+                config.displayPosition.renderRenderables(
+                    listOf(Renderable.fixedSizeLine(horizontalContainer(display, textAlignment.alignment), maxWidth)),
+                    posLabel = "Skill Progress")
+            }
+
+            else -> {}
+        }
+    }
+
+    private fun renderBar() {
+        val progress = if (barConfig.useTexturedBar.get()) {
+            val factor = (skillExpPercentage.toFloat().coerceAtMost(1f)) * 182
+            maxWidth = 182
+            Renderable.texturedProgressBar(factor,
+                Color(SpecialColour.specialToChromaRGB(barConfig.barStartColor)),
+                texture = barConfig.texturedBar.usedTexture.get(),
+                useChroma = barConfig.useChroma.get())
+
+        } else {
+            maxWidth = barConfig.regularBar.width
+            Renderable.progressBar(skillExpPercentage,
+                Color(SpecialColour.specialToChromaRGB(barConfig.barStartColor)),
+                Color(SpecialColour.specialToChromaRGB(barConfig.barStartColor)),
+                width = maxWidth,
+                height = barConfig.regularBar.height,
+                useChroma = barConfig.useChroma.get())
+        }
+
+        config.barPosition.renderRenderables(listOf(progress), posLabel = "Skill Progress Bar")
+    }
+
+    @SubscribeEvent
+    fun onProfileSwitch(event: PreProfileSwitchEvent) {
+        display = emptyList()
+        allDisplay = emptyList()
+        etaDisplay = emptyList()
+        skillExpPercentage = 0.0
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!isEnabled()) return
+        if (lastUpdate.passedSince() > 3.seconds) showDisplay = config.alwaysShow.get()
+
+        if (event.repeatSeconds(1)) {
+            allDisplay = formatAllDisplay(drawAllDisplay())
+            etaDisplay = drawETADisplay()
+        }
+
+        if (event.repeatSeconds(2)) {
+            update()
+            updateSkillInfo(activeSkill)
+        }
+    }
+
+    @SubscribeEvent
+    fun onLevelUp(event: SkillOverflowLevelupEvent) {
+        if (!isEnabled()) return
+        if (!config.overflowConfig.enableInChat) return
+        val skillName = event.skill.displayName
+        val oldLevel = event.oldLevel
+        val newLevel = event.newLevel
+        val skill = SkillAPI.storage?.skillData?.get(event.skill) ?: return
+        val goalReached = newLevel == skill.customGoalLevel && customGoalConfig.enableInChat
+
+        val rewards = buildList {
+            add("  §r§7§8+§b1 Flexing Point")
+            if (newLevel % 5 == 0)
+                add("  §r§7§8+§d50 SkyHanni User Luck")
+        }
+        val messages = listOf(
+            "§3§l▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬",
```

Comment:
maybe this could be a thing for stringUtils or ChatUtils so we have this for other places in the mod as well, maybe with the option to chnage the formatting codes at the start as well. (This is completely optional you dont ahve to do thi)